### DISCUSSION TITLE
Zephyr LoRaWAN Cannot use DR0 and DR1 on AS923 and AU915-928

### DISCUSSION THREAD

**Discussion by jumper385** [zephyrproject-rtos/zephyr#63405] [Q&A]
Hi All,

I am currently doing range tests for a project and I stumbled upon an issue with the LoRaMAC library. Despite trying to disable adaptive data rates using `lorawan_enable_adr(false)` and manually setting the data rate to `DR0` with `lorawan_set_datarate(LORAWAN_DR_0)`, the library persistently auto-sets the data rates to between `DR2` and `DR6`.

Upon digging deeper, I realized that the root of the issue lies in the region-specific override files provided by Semtech in the Zephyr LoRaMAC library. Specifically, the definition for `AS923_DWELL_LIMIT_DATARATE` as `DR_2` in `RegionAS923.h` seems to be the culprit.

```c
#define AS923_DWELL_LIMIT_DATARATE                  DR_2
```

My debugger's step-through pointed me to the `RegionAS923GetPhyParam` function where the data rate is being set to `DR2` (SF10BW125). However, when I tweaked the definition to `DR_0`, the received packet data rate displayed as `DR0` (SF12BW125), just as intended.

Interestingly, `RegionAS923.h` has the correct definition for the minimum data rate as `DR_0`:

```c
#define AS923_TX_MIN_DATARATE                       DR_0
```

Is there a way to disable the dwell limit data rate, and configure the `RegionAS923GetPhyParam` function to adopt `AS923_TX_MIN_DATARATE` instead, either via the project's config file `prj.conf` or directly within the project’s source files? Could there be a limitation with Zephyr causing this? Additionally, I know that there are regulatory requirements that LoRaWAN needs to comply with - specifically the 400ms dwell time limitation HOWEVER, I'm not aware of that regulation in Australia. I beleive this is the origin of the setting but i'm not sure what standard/regulatory reference it's complying with...

Kind Regards,
Henry

**Comment by JordanYates** [comment]
Based upon the conversation in this link (https://www.thethingsnetwork.org/forum/t/as923-dwell-time-limitation-for-australia/36601/8) even if the legal requirement for the 400ms dwell time was dropped, it can still be part of the Lorawan spec.
